src/samVisualize/graphGeneration/cfnTemplateGraphGenerator.ts

```diff
@@ -0,0 +1,152 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Graph, GraphObject } from './graph'
+import { yamlParse } from 'yaml-cfn'
+import { LinkTypes } from '../samVisualizeTypes'
+import * as _ from 'lodash'
+import { getLogger } from '../../shared/logger/logger'
+
+/**
+ * @param inputYaml A string representing a YAML template
+ * @returns A JavaScript object corresponding to the given YAML string, or undefined if the input is not valid YAML
+ */
+function yamlStringToObject(inputYaml: string): Record<string, any> | undefined {
```

Comment:
This is just calling yamlParse. Yet it is named yamlStringToObject. It is unnecessary entropy to have very different names for nearly the same thing. I would suggest naming this `yamlParseObject` or something, _anything_ starting with `yamlParseXX`, this helps discoverability and signals "these things are related".